docker/client.py

```diff
@@ -239,10 +241,14 @@ def _multiplexed_response_stream_helper(self, response):
         # Disable timeout on the underlying socket to prevent
         # Read timed out(s) for long running processes
         socket = self._get_raw_response_socket(response)
-        if six.PY3:
-            socket._sock.settimeout(None)
-        else:
+
+        # Depending on a few factors we might need to access
```

Comment:
I appreciate having a comment here is intended to be helpful, the vagueness of the description of what those factors are that's effecting the decision here isn't helping though. By removing the `if six.PY3` statement it seems less clear to me now what one of those factors are.

I prefer checking `six.PY3` but I also know what the arguments are for doing it with `hasattr`. If you would like to keep the `hasattr` then it needs pulling out into a little function with some specific explanation. Something like `_set_socket_timeout(socket)` that can then include a doc string that explains what the circumstances are for when we have to set the timeout in different places.